File: src/main/java/com/github/xiaolyuh/startup/ProjectStartupActivity.java
package com.github.xiaolyuh.startup;

import com.github.xiaolyuh.service.ConfigService;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.startup.ProjectActivity;
import kotlin.Unit;
import kotlin.coroutines.Continuation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ProjectStartupActivity implements ProjectActivity {

    @Override
    public @Nullable Object execute(@NotNull Project project, @NotNull Continuation<? super Unit> continuation) {
        ConfigService configService = ConfigService.Companion.getInstance(project);
        configService.tryInitConfig();
        return null;
    }

}

File: src/main/kotlin/com/github/xiaolyuh/startup/ProjectStartupActivity.kt
package com.github.xiaolyuh.startup

import com.github.xiaolyuh.service.ConfigService.Companion.getInstance
import com.intellij.openapi.project.Project
import com.intellij.openapi.startup.ProjectActivity

/**
 * @author yudong
 */
class ProjectStartupActivity : ProjectActivity {

    override suspend fun execute(project: Project) {
        val configService = getInstance(project)

        configService.tryInitConfig()
    }

}
